### PR TITLE
Support for AHT10 temperature and humidity sensor

### PR DESCRIPTION
**Pull request by gmasse**
## Description:

AHT10 is (yet another) low cost, high precision temperature and humidity I2C sensor.

**Related issue (if applicable):** https://github.com/esphome/feature-requests/issues/491

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#466

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/aht10/aht10.cpp [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    humidity = NAN;
+  } else {
+    humidity = (float) raw_humidity * 100.0 / 1048576.0;
+  }
+
+  if (this->temperature_sensor_ != nullptr) {
+    this->temperature_sensor_->publish_state(temperature);
+  }
+  if (this->humidity_sensor_ != nullptr) {
+    if (humidity == NAN)
```
Comment (glmnet) [inline esphome/components/aht10/aht10.cpp]: are you sure this works? I believe you must use isnan() here

**Review by glmnet — COMMENTED** [review]
LGTM, docs too

File: esphome/components/aht10/aht10.cpp (hunk 1 above)
Comment (gmasse) [reply esphome/components/aht10/aht10.cpp]: Good catch. Thanks @glmnet

**Comment by gmasse** [comment]
Bump.
This PR resolves https://github.com/esphome/feature-requests/issues/491.

**Comment by glmnet** [comment]
I’ll review it later. Thanks

**Comment by glmnet** [comment]
Ohhh already did. Ok let’s merge this one then

**Comment by gmasse** [comment]
Thanks @glmnet

**Pull request merged**